lib/watson-discovery-setup.js

```diff
@@ -245,53 +245,59 @@ WatsonDiscoverySetup.prototype.createDiscoveryCollection = function(params) {
-      if (!Number.isInteger(loadedDocs)) {
-        throw Error('Unexpected Discovery document_counts resulted in an unusable not-a-number.');
-      }
-      if (loadedDocs < docCount) {
-        console.log('Loading documents into Discovery collection.');
-        for (let i = 0; i < docCount; i++) {
-          const doc = docs[i];
-          const addDocParams = { file: fs.createReadStream(doc) };
-          Object.assign(addDocParams, params);
-          this.discoveryClient.addDocument(addDocParams, (err, data) => {
-            // Note: No promise. Just let these all run/log. Revisit this?
-            if (err) {
-              console.log('Add document error:');
-              console.error(err);
-            } else {
-              console.log('Added document:');
-              console.log(data);
-            }
-          });
-        }
+  return new Promise((resolve, reject) => {
+    console.log('Get collection to check its status.');
+    this.discoveryClient.getCollection(params, (err, data) => {
+      if (err) {
+        console.error(err);
+        return reject(err);
       } else {
-        console.log('Collection is already loaded with docs.');
-        /* For testing:
-         discovery.deleteCollection(params, (err, data) => {
-         console.log('Deleting collection for testing!!!!!!!!!!!!!!!!!!!!!!!!');
-         if (err) {
-         console.log(err);
-         }
-         });
-         */
+        console.log('Checking status of Discovery collection:', data.document_counts);
+        const numDocsLoaded = data.document_counts.available + data.document_counts.processing + data.document_counts.failed;
+        if ((! params.checkedForExistingDocs) && (numDocsLoaded > 0)) {
+          params.docsAlreadyLoaded = true;
+          params.checkedForExistingDocs = true;
+          console.log('Collection is already loaded with docs.');
+          resolve(params);
+        } else {
+          const docs = params.documents;
+          const docCount = docs.length;
+          params.checkedForExistingDocs = true;
+          var loadedDocs = params.loadedDocs;
+          var totalDocsToProcess = loadedDocs + docCount;
+          console.log('Loading documents into Discovery collection.');
+          for (let i = 0; i < docCount; i++) {
+            const doc = docs[i];
+            const addDocParams = { file: fs.createReadStream(doc) };
+            Object.assign(addDocParams, params);
+            this.discoveryClient.addDocument(addDocParams, (err, data) => {
+              if (err) {
+                console.log('Add document error:');
+                console.error(err);
+                params.loadedDocs += 1;
+              } else {
+                console.log('[' + params.loadedDocs + '] Added document: ' + data.document_id);
+                params.loadedDocs += 1;
+                if (params.loadedDocs === totalDocsToProcess) {
```

Comment:
unresolved promise when the if condition is false.  When it is true shouldn't it return the promise (?). Inside the callback unresolved promise on the error.